pkg/command/root.go

```diff
@@ -240,8 +247,41 @@ func newRootCmd() *cobra.Command {
 		// silencing usage for now as we are getting double usage from plugins on errors
 		SilenceUsage: true,
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) error {
-			// Sets the verbosity of the logger if TANZU_CLI_LOG_LEVEL is set
-			setLoggerVerbosity()
+			// Manually parse the flags
+			for i := 0; i < len(args); i++ {
+				arg := args[i]
+				if arg == "-v" || arg == "--verbose" {
```

Comment:
Cobra automatically verifies that there is a value set for the flag.
In fact, I don't believe this if statement will ever be true because by the time `PersistentPreRunE` is called, all flags have been parsed and removed from the list of args.

You can remove this whole loop

Reply:
So I have added this loop to parse args when plugin binaries are triggered.
1. For CLI commands; Yes args will be empty and this loop will not run
2. For Plugin commands; Since CLI doesn't parse args and sends all args to plugin this loop will run to verify if verbose flag is set by the user and then passes to plugin.

Reply:
Thanks added the check